plugins/antiflood.ts

```diff
@@ -0,0 +1,35 @@
+import { createPlugin } from "../core/plugins.ts";
+import { Queue } from "../deps.ts";
+
+interface AntiFloodFeatures {
+  options: {
+    /** Milliseconds to wait between dispatching private messages.
+     *
+     * Defaults to 0 milliseconds (no delay) */
+    floodDelay?: number;
+  };
+}
+
+export default createPlugin("antiflood", [])<AntiFloodFeatures>(
+  (client, options) => {
```

Comment:
```suggestion
  (client, options) => {
    if (!options.floodDelay) return;
```

just in order to prevent hooking send() if not needed.

Reply:
I think this is a great change, saves some resources.

I've enhanced the check on `floodDelay` here to remove the other check we discussed.